### PR TITLE
Update c# language version to 9

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

**Comment by peppy** [comment]
I've applied remaining changes in line with what was done on the framework side, plus bumped the framework here.

**Comment by peppy** [comment]
Painful. Wanna make a PR reverting that if you're able to?

**Comment by bdach** [comment]
Done (https://github.com/ppy/osu-framework/pull/5266).

**Comment by bdach** [comment]
Bumped framework again (manually, so buyer beware), as well as fixed the failures from the framework breaking change to texture stores in [a83c45b](https://github.com/ppy/osu/pull/18830/commits/a83c45b50e4f126bfa218d4bd8e4d8a2bf1e5145). @frenzibyte would appreciate a double-check of the latter if/when able, if it matches the changes you had queued.

**Review by smoogipoo — DISMISSED** [review]

File: osu.sln.DotSettings [hunk 1]
```diff
@@ -7,19 +7,22 @@
...
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ArrangeRedundantParentheses/@EntryIndexedValue">WARNING</s:String>
 	<s:Boolean x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ArrangeRedundantParentheses/@EntryIndexRemoved">True</s:Boolean>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ArrangeTypeMemberModifiers/@EntryIndexedValue">WARNING</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=ArrangeTypeModifiers/@EntryIndexedValue">WARNING</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=AssignedValueIsNeverUsed/@EntryIndexedValue">HINT</s:String>
 	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=AssignmentIsFullyDiscarded/@EntryIndexedValue">DO_NOT_SHOW</s:String>
-	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=AssignNullToNotNullAttribute/@EntryIndexedValue">WARNING</s:String>
-	<s:String x:Key="/Default/CodeInspection/Highlighting/InspectionSeverities/=AsyncVoidMethod/@EntryIndexedValue">WARNING</s:String>
```
Comment (bdach) [inline osu.sln.DotSettings:22]: How were these dotsettings changes made? Looks like the file may have been copy-pasted verbatim from the framework directory to the game's, but in the meantime this subtly changed the set of enabled inspections (like the "async void" one above).

Comment (bdach) [reply osu.sln.DotSettings:22]: I have changes queued that revert all inspection downgrades that happened as a result of the procedure you described, just want to run them past inspectcode again to make sure I haven't missed anything.

**Review by frenzibyte — APPROVED** [review]

**Pull request merged**